ランディングページの新規作成

HrsUed opened the pull request:
# What
ユーザが最初に訪問するページを実装する

# Why
必須画面であるため

HrsUed committed:
Create Pull Request

HrsUed committed:
bootstapとfont-awesomeのインストール

HrsUed committed:
アプリケーションレイアウトを初期化
・scssに変更
・リセットファイルの導入
・bootstrapとfont-awesomeのインポート

HrsUed committed:
welcomeコントローラの生成

HrsUed committed:
welcome#indexをrootに変更

HrsUed committed:
ウェルカムページの実装

HrsUed committed:
プロジェクトREADMEの修正

HrsUed commented on app/assets/stylesheets/_reset.scss:
```diff
@@ -0,0 +1,126 @@
+    font-weight:inherit;
+}
+/*to enable resizing for IE*/
+input,
```
> Unexpected duplicate selector "input,
> textarea,
> select", first used at line 108 (no-duplicate-selectors)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

IE用の追加スタイルであり、テンプレートであるため修正不要

HrsUed commented on app/assets/stylesheets/_reset.scss:
```diff
@@ -0,0 +1,126 @@
+    border:0;
+}
+/*
+    TODO think about hanlding inheritence differently, maybe letting IE6 fail a bit...
```
> "inheritence" is a misspelling of "inheritance"
> [link](https://sideci.com/gh/132450075/pull_requests/12)

修正済み

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,241 @@
+
+  .StaticLoggedOutHomePage-content {
+    background-color: #fff;
+    @extend .flex-md-row-reverse;
```
> Unexpected unknown at-rule "@extend" (at-rule-no-unknown)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

修正済み

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,241 @@
+
+            input {
+              width: 100%;
+              @extend .form-control;
```
> Unexpected unknown at-rule "@extend" (at-rule-no-unknown)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

Bootstrapのクラスを読み込みしており、問題なし

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,241 @@
+            }
+          }
+
+          .LoginForm-username {
```
> Unexpected empty block (block-no-empty)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

未定義のクラスを許可する

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,241 @@
+            }
+          }
+
+          .LoginForm-submit {
```
> Unexpected empty block (block-no-empty)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

未定義のクラスを許可する

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,241 @@
+  }
+}
+
+.StaticLoggedOutHomePage-signupSubtitle {
```
> Expected selector ".StaticLoggedOutHomePage-signupSubtitle" to come before selector ".StaticLoggedOutHomePage .StaticLoggedOutHomePage-content .StaticLoggedOutHomePage-utilityBlock .StaticLoggedOutHomePage-signupBlock .StaticLoggedOutHomePage-noSignupForm .StaticLoggedOutHomePage-signupSubtitle" (no-descending-specificity)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

Twitterとの整合性は取れているが、一旦親子関係のクラス側に集約する対応とする。

HrsUed committed:
SideCIによる指摘修正

Hermiltonian commented:
・ブラウザ最大化時のページ
https://gyazo.com/0005acd4f65400e95485adf5ee8bc32a

・注意メッセージ読了時の動作
https://gyazo.com/dd58ddb0cb82b3abadf7ab36503bd39f

・レスポンシブデザインの確認（幅変更）
https://gyazo.com/1b285f271f6bedd65a7f4a2158cd40c7

・レスポンシブデザインの確認（高さ変更）

HrsUed committed:
ウェルカムページ用jsの修正
・コメント追記
・コンテンツの高さに下限値を設定追加

HrsUed committed:
本家に近づけるため、Bootstrapの.container, .rowを削除

HrsUed committed:
・ビューから.container, .row削除したことによるscssへの反映
・ネスト誤りの修正
・不要な空行の削除、必要な空行の挿入

HrsUed committed:
バグ修正：コンテンツ高の制御ロジックを修正

Hermiltonian commented:
＜ビューのキャプチャ＞
・ブラウザ最大化時のページ
https://gyazo.com/ae300c52b2a45403bec15d23ee0a7653

・注意メッセージ読了時の動作
https://gyazo.com/f7b0bc8b4fc88287f7d63527438999bf

・レスポンシブデザインの確認（幅変更）
https://gyazo.com/1b5ffcbb8bbbc41693ae7b4e77777714

ta910 commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,265 @@
+.StaticLoggedOutHomePage {
+  min-height: 100%;
+  // height: 100vh;
```
消しておきましょう

HrsUed replied:
修正しました。

ta910 commented:
全体的に良いです！
ただ、scssが少し読みにくく、空行が多すぎる気がします。
ブロックごとや、要素ごとにまとめてあるような構造が読みやすいので、少し空行を入れる場所を工夫してみてください

HrsUed committed:
レビュー指摘修正
・scssの不要なコメント削除
・scssの空行を調節してブロックのグルーピングと分離でメリハリ付け

HrsUed committed:
jsのスペルミス修正

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,206 @@
+              @extend .form-control;
+            }
+          }
+          .LoginForm-username {}
```
> Unexpected empty block (block-no-empty)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

保守性を考慮し、空のスタイルを許可する

HrsUed commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,206 @@
+              a.forgot {color: #aab8c2;}
+            }
+          }
+          .LoginForm-submit {}
```
> Unexpected empty block (block-no-empty)
> [link](https://sideci.com/gh/132450075/pull_requests/12)

今後の拡張性を考慮し、空のスタイルを許可する

quelmecabj commented on app/assets/stylesheets/_welcome.scss:
```diff
@@ -0,0 +1,206 @@
+      .twitterIcon-bird {
+        max-width: none;
+        position: absolute;
+        @media only screen and (min-width: 800px) {height: 140vh; right: -80vh; top: -20vh;}
```
可読性を上げるため、以下のように書き方を統一しましょう！
@media only screen and (min-width: 800px) {
  height: 140vh;
  right: -80vh;
  top: -20vh;
}

また、mediaクエリをよく使うのであれば、mixinで定義してもいいかもしれないですね！

HrsUed replied:
記載統一しました。
また、アドバイスありがとうございます。mixinは今後他のビュー作成で乱立した際の、リファクタリングToDoにしたいと思いますので、今回は勘弁してください。。。

quelmecabj reviewed (commented):
お疲れ様です！
コメントいたしました。
よろしくお願いいたします！

HrsUed committed:
レビュー指摘修正
・スタイルを1行で記載することを排除して記載方法を統一
・漏れていたメディアタイプの記載を追記

HrsUed committed:
ssl環境未セットアップのため設定を解除する

HrsUed committed:
本番環境へのデプロイ時エラーに対応するため、以下変更。
エラー：Invalid option key: raise_on_unfiltered_parameters=
・Rails.application.config.action_controller.raise_on_unfiltered_parameters = trueをコメントアウト

HrsUed committed:
アプリケーションREADMEに本番環境へのリンクを追記
